app/containers/Vaults/index.js

```diff
@@ -479,10 +493,20 @@ const VaultsWrapper = (props) => {
       vaultKey = `${vault.address}-eth`;
     }
     let banner = null;
+    let decimals = 18;
+    if (vault && vault.token && vault.token.decimals > 0) {
+      if (vault.token.decimals > 6) {
+        decimals = vault.token.decimals - 4;
+      } else if (vault.tokens.decimals >= 4) {
```

Comment:
> } else if (vault.tokens.decimals >= 4) {

Typo here, should be
`vault.token.decimals`

If you can refactor this as a function so that it does not repeat

Reply:
done